src/server/dfly_main.cc

```diff
@@ -701,6 +711,46 @@ void PrintBasicUsageInfo() {
   std::cout << endl;
 }
 
+void ParseFlagsFromEnv() {
+  if (getenv("DFLY_PASSWORD")) {
+    LOG(WARNING)
+        << "DFLY_PASSWORD environment variable is being deprecated in favour of DFLY_requirepass";
+  }
+  // Allowed environment variable names that can have
+  // DFLY_ previx, but don't necessirily have an ABSL flag created
+  std::set<std::string_view> allowed_environment_flag_names = {"dev_env", "password"};
+  const auto& flags = absl::GetAllFlags();
+  for (char** env = environ; *env != nullptr; env++) {
+    constexpr string_view kPrefix = "DFLY_";
+    string_view environ_var = *env;
+    if (absl::StartsWith(environ_var, kPrefix)) {
+      // Per 'man environ', environment variables are included with their values
+      // in the format "name=value". Need to strip them apart, in order to work with flags object
+      pair<string_view, string_view> environ_pair = absl::StrSplit(
+          absl::AsciiStrToLower(absl::StripPrefix(environ_var, kPrefix)), absl::MaxSplits('=', 1));
+      const auto& [flag_name, flag_value] = environ_pair;
+      if (allowed_environment_flag_names.find(flag_name) != allowed_environment_flag_names.end()) {
```

Comment:
By switching to `absl::flat_hash_set` you'll have the `.contains()` method which is much less verbose for this case :)
(`std::set` also has `.contains()`, but it's only C++20 and we're using C++17)